tool/tsh/proxy.go

```diff
@@ -207,12 +210,25 @@ func sshProxy(tc *libclient.TeleportClient, params sshProxyParams) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	keysDir := profile.FullProfilePath(tc.Config.KeysDir)
-	knownHostsPath := keypaths.KnownHostsPath(keysDir)
+
+	// Get knownhosts, ssh certificate, and identity from the tsh profile.
+	profile, err := profile.FromDir(tc.Config.KeysDir, "")
+	if err != nil {
+		if trace.IsNotFound(err) {
+			return trace.BadParameter("No current profile detected, are you logged in?")
+		}
+		return trace.Wrap(err)
+	}
+
+	// If we are using a jump host, then the desired cluster name might
+	// be different from the current cluster of the tsh profile.
+	profile.SiteName = params.clusterName
 
 	args := []string{
 		"-A",
-		"-o", fmt.Sprintf("UserKnownHostsFile=%s", knownHostsPath),
+		"-o", fmt.Sprintf("UserKnownHostsFile=%s", profile.KnownHostsPath()),
+		"-o", fmt.Sprintf("CertificateFile=%s", profile.SSHCertPath()),
+		"-o", fmt.Sprintf("IdentityFile=%s", profile.UserKeyPath()),
```

Comment:
I added this and the `-v` below as small quality of life fixes which I wanted to include after debugging the issue in https://github.com/gravitational/teleport/pull/13956

Basically, with this change, the SSH certificate and identity files in `~/.tsh` will be added to the identity chain, which currently only consists of `ssh-agent` and any configuration in their `~/.ssh/config`. As a result, even with no configured Certificate/Identity and no `ssh-agent`, `tsh proxy ssh` should still succeed, as expected.

Additionally, In a jump host scenario, we re-issue a new leaf certificate for the jump host after https://github.com/gravitational/teleport/pull/12848. We depend on it being added to the `ssh-agent`, but again if the user doesn't have an `ssh-agent` running, with this change it will still find and use the correct certificate.